pdg/src/info.rs

```diff
@@ -1,24 +1,32 @@
-use crate::graph::{Graph, NodeId, NodeKind};
+use crate::graph::{Graph, Node, NodeId, NodeKind};
 use crate::Graphs;
-use std::collections::HashMap;
+use rustc_middle::mir::Field;
+use std::cmp::max;
+use std::collections::hash_map::Entry;
+use std::collections::{HashMap, HashSet};
 use std::fmt::{self, Debug, Display, Formatter};
 
 /// Information generated from the PDG proper that is queried by static analysis.
 ///
 /// Includes information about what kinds of [`Node`]s the [`Node`] flows to,
-/// and eventually will also include its ability to be used as a `&mut`.
+/// as well as its ability to be used as a `&mut`.
 ///
 /// [`Node`]: crate::graph::Node
 #[derive(Hash, Clone, PartialEq, Eq, Debug)]
 pub struct NodeInfo {
     flows_to: FlowInfo,
+    unique: bool,
```

Comment:
documenting `unique` and what it means would be helpful

Reply:
what if I add in the parenthetical
`as well as its ability to be used as a `&mut` (signified by 'unique')`?

Reply:
i think it should be added above the attribute itself, not in the general struct documentation